library/ssl_tls.c

```diff
@@ -1216,6 +1227,16 @@ int mbedtls_ssl_derive_keys( mbedtls_ssl_context *ssl )
                                   mac_key_len, transform->keylen,
                                   iv_copy_len );
     }
+
+    if( ssl->conf->f_export_keys_ext != NULL )
```

Comment:
OK. But the single exit point was to postpone the erasure of secret data, so I think they are very well connected.

Reply:
Yes, they are very well connected and doing it was motivated by the change you mention. But it is a valuable addition on its own and doesn't change behaviour: it is a refactoring that you did in preparation for the real change. We make these changes usually in a separate, preceding commit.

I suggested to put them in separate commits these two changes, because I think that it is easier to look at them separately and easier to comprehend what they do if they are presented in separate commits.

Reply:
ok